presentation/src/main/java/com/example/pinterbest/presentation/adapters/CreatorsAdapter.kt

```diff
@@ -0,0 +1,55 @@
+package com.example.pinterbest.presentation.adapters
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+import com.example.pinterbest.presentation.R
+import com.example.pinterbest.presentation.databinding.ViewHolderCreatorBinding
+import com.example.pinterbest.presentation.models.ProfileViewData
+
+class CreatorsAdapter :
+    RecyclerView.Adapter<CreatorsAdapter.ViewHolder>() {
+    private var profiles = listOf<ProfileViewData>()
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val binding = ViewHolderCreatorBinding.inflate(
+            LayoutInflater.from(parent.context),
+            parent,
+            false
+        )
+        return ViewHolder(binding)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(profiles[position])
+    }
+
+    override fun getItemCount() = profiles.size
+
+    fun updateList(profilesList: List<ProfileViewData>) {
+        profiles = profilesList
+        notifyDataSetChanged()
+    }
+
+    class ViewHolder(private val binding: ViewHolderCreatorBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+
+        fun bind(profile: ProfileViewData) {
+            binding.name.text = profile.username
+            binding.pinsCount.text = profile.pinsCount.toString()
+            binding.followers.text = profile.followers.toString()
+            binding.followings.text = profile.following.toString()
+            val url = BASE_URL_IMAGES + profile.avatarLink
+            Glide.with(binding.imageView.context)
+                .load(url)
+                .centerCrop()
+                .error(R.drawable.ic_error)
+                .into(binding.imageView)
+        }
+    }
+
+    companion object {
+        const val BASE_URL_IMAGES = "https://pinterbest-bucket.s3.eu-central-1.amazonaws.com/"
```

Comment:
this URL **must** be already included in `profile.avatarLink` field. It's not responsibility of adapters to decide which base url to use for images. For example you can concat avatarLink with BASE_URL_IMAGES in Repository

Reply:
Fixed here https://github.com/enginire4ek/Pinterbest/pull/13/commits/7e75b3b148c4925e3a8f02f2ffc3c27aaf25c304